src/test/java/plugins/WorkflowPluginTest.java

```diff
@@ -135,6 +137,8 @@
         build.shouldContainsConsoleOutput("Building version 1.0-SNAPSHOT");
         jenkins.restart();
         // Default 120s timeout of Build.waitUntilFinished sometimes expires waiting for RetentionStrategy.Always to tick (after initial failure of CommandLauncher.launch: EOFException: unexpected stream termination):
+        System.out.println("Waiting for slave to come online...");
+        Thread.sleep(10000); //Jenkins needs some extra time to wake up before we can query the api
```

Comment:
`waitUntilOnline` checks `slave.isOnline` which in essence is `!getJson().get("offline").asBoolean()`

It works, but only when Jenkins is fully online apparently. It seems like `jenkins.restart()` isn't waiting long enough or something. I tried various tweaks but I just ended up tumbling further and further down the rabbit hole, so I decided to dig myself up before I spent too much time on something I wasn't supposed to spend time on.